lib/coll.gi

```diff
@@ -976,7 +976,7 @@ InstallGlobalFunction( IteratorByFunctions, function( record )
       Error( "<record> must be a record with components `NextIterator',\n",
              "`IsDoneIterator', and `ShallowCopy'" );
     fi;
-    filter:= IsIteratorByFunctions and IsMutable;
+    filter:= IsIteratorByFunctions and IsStandardIterator and IsMutable;
```

Comment:
Doesn't this mean that line 968 above in
```
DeclareSynonym( "IsIteratorByFunctions",
     IsIteratorByFunctionsRep and IsIterator );
```
should be updated y adding `IsStandardIterator` as well?